app/src/main/java/org/mozilla/fenix/gleanplumb/MessagingStorage.kt

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.gleanplumb
+
+import android.content.Context
+import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONObject
+import org.mozilla.experiments.nimbus.GleanPlumbInterface
+import org.mozilla.experiments.nimbus.GleanPlumbMessageHelper
+import org.mozilla.experiments.nimbus.internal.FeatureHolder
+import org.mozilla.experiments.nimbus.internal.NimbusException
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppAction.MessagingAction.UpdateMessages
+import org.mozilla.fenix.nimbus.Messaging
+import org.mozilla.fenix.nimbus.StyleData
+
+class MessagingStorage(
```

Comment:
I was thinking about `NimbusMessagingProvider`
I'll cover it on https://github.com/mozilla-mobile/fenix/pull/24227

Reply:
I'd recommend ending it with "Storage" because that's a naming convention we already use for features that work in similar ways (persistent storage); I think the pattern is now to fallback to "Provider" when the storage is complex and the storage has it's own inner storage layer.

Reply:
👍🏽